### ISSUE TITLE
ActivityNet 1.3 Dataloader

### ISSUE THREAD

**Issue by sauradip** [MichiganCOG/A2CL-PT#6]
Can you upload the dataloader for ActivityNet 1.3 as most of the WTAL codes have skipped this, and hyperparams affect the WTAL setting, so kindly upload the hyperparams as well.

THanks in advance

**Comment by kylemin** [comment]
Hi,
Thank you for your interest.
Please refer to this previous [discussion](https://github.com/MichiganCOG/A2CL-PT/issues/4). We used the same dataloader for both datasets, so you do not need to change anything. Also, I believe that all the hyperparameters are already described in the paper and that there are other people who have successfully reproduced the reported results.
Please let me know if there are hyperparameters that I forgot to make public.
Kyle

**Comment by sauradip** [comment]
> Hi,
> Thank you for your interest.
> Please refer to this previous [discussion](https://github.com/MichiganCOG/A2CL-PT/issues/4). We used the same dataloader for both datasets, so you do not need to change anything. Also, I believe that all the hyperparameters are already described in the paper and that there are other people who have successfully reproduced the reported results.
> Please let me know if there are hyperparameters that I forgot to make public.
> Kyle

Thanks for your comment. I managed to get the dataloader working for training. For validation what will be the "feature_rate" used in detection, i saw the feature_rate for THUMOS14 is 1.56. what will be for AN1.3 ?

**Comment by sauradip** [comment]
> Hi,
> Thank you for your interest.
> Please refer to this previous [discussion](https://github.com/MichiganCOG/A2CL-PT/issues/4). We used the same dataloader for both datasets, so you do not need to change anything. Also, I believe that all the hyperparameters are already described in the paper and that there are other people who have successfully reproduced the reported results.
> Please let me know if there are hyperparameters that I forgot to make public.
> Kyle

Hi, 

I ran into same issue as the author in the discussion mentioned. I am getting very poor MAP. Second thing is the discussion mentions something about tfilter, and the checkpoint you uploaded in the link does not have the weight of the tfilter. So how was the training done ?

**Comment by kylemin** [comment]
Please check that you downloaded the correct one. The name of the weight file should be 'best1-act.pt'. I checked it again, and it works just fine for me. It obviously has the weights of the tfilter. If you still have problems, please let me know.

**Comment by sauradip** [comment]
> Please check that you downloaded the correct one. The name of the weight file should be 'best1-act.pt'. I checked it again, and it works just fine for me. It obviously has the weights of the tfilter. If you still have problems, please let me know.

Thanks , i checked now, i downloaded the correct version now ! 
Can you answer the second query regarding the poor Detection Accuracy inspite of having better classification accuracy ? 
How many iterations did you train this checkpoint ?

**Comment by kylemin** [comment]
That is interesting. I think that there might be an error in the classification.py? Let me look into it. 175k iterations, as we mentioned in our paper!

**Comment by sauradip** [comment]
I trained myself for ANet1.3 for around 30k iterations, classification accuracy around 88% with MIOU of 0.1 stilll at 0.01 but good thing is now i got the classes correct. With your version, although the classification score was high i saw that the model was overfitting with 3-4 classes constantly repeated.

**Comment by kylemin** [comment]
Well, now I am confused. Can you elaborate more about what your problem is? Are you saying that the pretrained model performs bad in a localization task?

**Comment by sauradip** [comment]
CHeck this 
https://github.com/MichiganCOG/A2CL-PT/issues/4#issuecomment-705766547
Here this person also got poor performance. Can you recheck the performance of the shared checkpoint for localization ? The localization is not coming as reported from your shared file. 

THe problem in short is that the checkpoints weights are not giving reported results additionally try printing the predicted results  and check. I got few classes repreated all over the validation video. 

Since THUMOS is small i dont think this has problem , may be for ANet since you have not provided dataloader , my bet is that the dictionary for your classes ( self.cnames ) might have different indexing as compared to us. But again thanks a ton for discussing this issue. You can reply me here : s.nag@surrey.ac.uk instead of this thread if you wish.

**Comment by kylemin** [comment]
I rechecked the performance of the released checkpoint and it works as expected for localization.. Let me check the indices and the dataloader.
